extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java

```diff
@@ -149,6 +153,19 @@ private static VertxOptions convertToVertxOptions(VertxConfiguration conf) {
         return options;
     }
 
+    private static int calculateDefaultIOThreads() {
```

Comment:
We also need to reduce the number of verticle deployed.

Event loops are created lazily (but it's always good to cap). when deploying a verticle, an event loop is assigned to it, which trigger the creation if the max is not reached. If you have more verticle than event loops, some verticle would share the event loop.